db/1.9/modules/ROOT/pages/database-stored-procedure.adoc

```diff
@@ -0,0 +1,211 @@
+= Execute Stored Procedures Examples - Mule 4
+
+Anypoint Connector for Database (Database Connector) Stored Procedure operation invokes a stored procedure on the database. When the stored procedure returns one or more ResultSet instances, streaming is applied automatically to avoid preemptive consumption of the results, which can lead to performance and memory issues.
+
+The following examples illustrates callable statements and dates on stored procedures, and how to invoke stored procedures by configuring Input, Output, and In Out parameters.
+
+== Callable Statements in Stored Procedures
+
+Database Connector supports the use of callable statements with the format:
+
+`{ call procedureName(:param1, :param2, ..., :paramN) }`
+
+Where `:paramN` matches the `Nth` parameter in the stored procedure declaration. Make sure that no whitespace is between the name of the stored procedure and the first parentheses.
+
+The following example illustrates an Oracle database that was initialized when creating a table named `SYSTEM.employees` and stored procedure named `createEmployee`:
+
+[source,xml,linenums]
+----
+    CREATE TABLE SYSTEM.employees(
+        employee_id INTEGER GENERATED BY DEFAULT AS IDENTITY,
+        employee_name VARCHAR2(100),
+        employee_age INTEGER,
+        employee_birthday TIMESTAMP,
+        PRIMARY KEY(employee_id)
+    );
+
+
+    CREATE PROCEDURE createEmployee(e_name VARCHAR2, e_age NUMBER, e_birth_date DATE) AS
+    BEGIN
+        INSERT INTO SYSTEM.employees(employee_name, employee_age, employee_birthday) VALUES(e_name, e_age, e_birth_date);
+    END;
+----
+
+Because Database Connector does not support named parameters in callable statements, attempting to match parameters by name does not work, for example:
+
+`{ call createEmployee(employee_age => :age, e_birth_date => :date, e_name => :name) }`
+
+You must provide the parameters in the appropriate order, for example:
+
+`{ call createEmployee(e_name => :name, employee_age => :age, e_birth_date => :date) }`.
+
+
```

Comment:
Add the following best practices and recommendations which avoid some issues/confusions we had with customers:

- When passing parameters to an SP please:
  - Place all the parameters the SP has (even Optional parameters -the ones with default values)
  - Put them in the same order as the SP has them defined
  - Parametrize all of them (meaning, do not hardcode strings, numbers, etc.)
  
- When giving value to the parameters in the 'Input Parameters' box please:
   - Define each parameter name within single quotes (`'paramName': value`)
   - Default Value: Either use DW or a variable since these are not supported - so if you pass null, empty string, and so the SP's default value won't be pushed by Mule